src/serialization/DefaultSerializers.ts

```diff
@@ -428,3 +435,158 @@ export class UuidSerializer implements Serializer<UUID> {
+    }
+
+    write(output: DataOutput, hzLocalTime: LocalTime): void {
+        output.writeByte(hzLocalTime.hour);
+        output.writeByte(hzLocalTime.minute);
+        output.writeByte(hzLocalTime.second);
+        output.writeInt(hzLocalTime.nano);
+    }
+}
+
+/** @internal */
+export class LocalDateTimeSerializer implements Serializer<LocalDateTime> {
+
+    id = -53;
+
+    read(input: DataInput): LocalDateTime {
+        const year = input.readInt();
+        const month = input.readByte();
+        const date = input.readByte();
+
+        const hour = input.readByte();
+        const minute = input.readByte();
+        const second = input.readByte();
+        const nano = input.readInt();
+
+        return LocalDateTime.new(year, month, date, hour, minute, second, nano);
+    }
+
+    write(output: DataOutput, hzLocalDateTime: LocalDateTime): void {
+        output.writeInt(hzLocalDateTime.localDate.year);
+        output.writeByte(hzLocalDateTime.localDate.month);
+        output.writeByte(hzLocalDateTime.localDate.date);
+
+        output.writeByte(hzLocalDateTime.localTime.hour);
+        output.writeByte(hzLocalDateTime.localTime.minute);
+        output.writeByte(hzLocalDateTime.localTime.second);
+        output.writeInt(hzLocalDateTime.localTime.nano);
+    }
+}
+
+/** @internal */
+export class OffsetDateTimeSerializer implements Serializer<OffsetDateTime> {
+
+    id = -54;
+
+    read(input: DataInput): OffsetDateTime {
+        const year = input.readInt();
+        const month = input.readByte();
+        const date = input.readByte();
+
+        const hour = input.readByte();
+        const minute = input.readByte();
+        const second = input.readByte();
+        const nano = input.readInt();
+
+        const offsetSeconds = input.readInt();
+
+        return OffsetDateTime.new(year, month, date, hour, minute, second, nano, offsetSeconds);
+    }
+
+    write(output: DataOutput, hzOffsetDateTime: OffsetDateTime): void {
+        output.writeInt(hzOffsetDateTime.localDateTime.localDate.year);
+        output.writeByte(hzOffsetDateTime.localDateTime.localDate.month);
+        output.writeByte(hzOffsetDateTime.localDateTime.localDate.date);
+
+        output.writeByte(hzOffsetDateTime.localDateTime.localTime.hour);
+        output.writeByte(hzOffsetDateTime.localDateTime.localTime.minute);
+        output.writeByte(hzOffsetDateTime.localDateTime.localTime.second);
+        output.writeInt(hzOffsetDateTime.localDateTime.localTime.nano);
+
+        output.writeInt(hzOffsetDateTime.offsetSeconds);
+    }
+}
+
+/** @internal */
+export class BigDecimalSerializer implements Serializer<BigDecimal> {
+
+    id = -27;
+
+    read(input: DataInput): BigDecimal {
+        const body = input.readByteArray();
+        const scale = input.readInt();
+
+        return new BigDecimal(fromBufferAndScale(body, scale));
+    }
+
+    write(output: DataOutput, bigDecimal: BigDecimal): void {
```

Comment:
Have you verified we can read the different kinds of values written by the Node.js client in Java? I mean, yes I see that we test some number in serialization tests but are things like negative values etc. tested?

Reply:
it should be in DefaultSerializersLiveTest.js I guess?

Reply:
I mean am I supposed to write it there, they are not written yet

Reply:
@mdumandag Added tests to `DefaultSerializersLiveTest.js`, and fixed your other comments(other than the ones that I am not sure what to do) you can rereview this pr.

Also do you think `static new()` constructors are ok, or should I rename it to `from()` ?

Reply:
I think `from` might sound better and be less confusing (`new XX()` and `XX.new()` look alike), but I don't have a strong opinion here